src/test/perl/path.t

```diff
@@ -108,9 +108,9 @@ sub verify_exception
     } else {
         ok(! $ec_check->error(), "Error reset after $msg");
     };
-    if ($noreset) {
+    if ($noreset && $mc->{fail}) {
```

Comment:
You are right that `$noreset` was not honored when '$mc->{fail}` was undefined. Should be fixed now and I really think that nothing is changed in the logic, just meaningless `like()` are not executed. @stdweird do you agree?